noark-extraction-validator/src/main/java/com/documaster/validator/validation/noark53/validators/NoarkXMLHandler.java

```diff
@@ -0,0 +1,225 @@
+				.map(ex -> {
+
+					BaseItem item = new BaseItem();
+
+					item.add("Schema", schema.toString());
+					item.add("Type", ex.getElementType());
+					ex.getIdFields().forEach((k, v) -> item.add(k.toString(), v));
+					item.add("Line", ex.getLineNumber() != -1 ? ex.getLineNumber() : null);
+					item.add("Column", ex.getColumnNumber() != -1 ? ex.getColumnNumber() : null);
+					item.add("Message", ex.getMessage());
+
+					return item;
+				}).collect(collectingAndThen(toList(), Collections::unmodifiableList));
+	}
+
+	@Override
+	public void reset() {
+
+		exceptions.clear();
+		characters.setLength(0);
+		elementType = null;
+		Stream.of(NoarkReferenceField.values()).forEach(v -> idFields.put(v, null));
+	}
+
+	@Override
+	public void startElement(String uri, String localName, String qName, Attributes attributes) {
+
+		characters.setLength(0);
+		setElementType(qName, attributes);
+	}
+
+	@Override
+	public void characters(char buf[], int offset, int length) {
+
+		characters.append(new String(buf, offset, length));
+	}
+
+	@Override
+	public void endElement(String uri, String localName, String qName) {
+
+		for (NoarkReferenceField field : NoarkReferenceField.values()) {
+			if (qName.equalsIgnoreCase(field.getName())) {
+				setFieldValue(field, characters.toString().trim());
+				break;
+			}
+		}
+	}
+
+	@Override
+	public void warning(SAXParseException exception) {
+
+		exceptions.add(new NoarkSAXParseException(exception, elementType, idFields));
+	}
+
+	@Override
+	public void error(SAXParseException exception) {
+
+		exceptions.add(new NoarkSAXParseException(exception, elementType, idFields));
+	}
+
+	@Override
+	public void fatalError(SAXParseException exception) {
+
+		exceptions.add(new NoarkSAXParseException(exception, elementType, idFields));
+	}
+
+	private static class NoarkSAXParseException extends SAXParseException {
+
+		private Map<NoarkReferenceField, String> idFields;
+		private String elementType;
+
+		NoarkSAXParseException(SAXParseException ex, String elementType, Map<NoarkReferenceField, String> idFields) {
+
+			super(ex.getMessage(), ex.getPublicId(), ex.getSystemId(), ex.getLineNumber(), ex.getColumnNumber());
+			this.elementType = elementType;
+			this.idFields = idFields;
+		}
+
+		String getElementType() {
+
+			return elementType;
+		}
+
+		Map<NoarkReferenceField, String> getIdFields() {
+
+			return idFields;
+		}
+	}
+
+	public enum Schema {
+		PACKAGED, NOARK;
+	}
+
+	private enum NoarkElementType {
+
+		ARKIV, ARKIVSKAPER, ARKIVDEL, KLASSIFIKASJONSSYSTEM, KLASSE, SAKSMAPPE, MAPPE, JOURNALPOST, MOETEMAPPE,
+		REGISTRERING, SAKSPART, BASISREGISTRERING, MOETEDELTAKER, KORRESPONDANSEPART, AVSKRIVNING, DOKUMENTFLYT,
+		DOKUMENTBESKRIVELSE, MOETEREGISTRERING, DOKUMENTOBJEKT, KONVERTERING, KRYSSREFERANSE, MERKNAD, KASSASJON,
+		UTFOERTKASSASJON, SLETTING, SKJERMING, GRADERING, PRESEDENS, ELEKTRONISKSIGNATUR, ENDRING, JOURNALHODE,
+		OURNALREGISTRERING, ENDRINGSLOGG, OFFENTLIGJOURNAL, LOEPENDEJOURNAL;
```

Comment:
The NoarkElementType enum represents all non-leaf XML element names that can be found in _arkivstruktur.xml_. _arkivstruktur.xsd_ does not define an element named _journalregistrering_. It defines _journalpost_.